composer/profiler/dataloader_profiler.py

```diff
@@ -22,9 +23,9 @@ class ProfiledDataLoader(WrappedDataLoader):
         name (str): The name for the dataloader.
     """
 
-    def __init__(self, profiler: Profiler, dataloader: DataLoader, name: str) -> None:
+    def __init__(self, state: State, dataloader: DataLoader, name: str) -> None:
         super().__init__(dataloader)
-        self._marker = profiler.marker(f"dataloader/{name}", categories=["dataloader"])
+        self._marker = state.profiler.marker(f"dataloader/{name}", categories=["dataloader"])
```

Comment:
I'd be surprised if this is passing pyright since profiler is an optional member of state.